Mage/src/main/java/mage/game/GameImpl.java

```diff
@@ -2733,6 +2735,28 @@ protected boolean checkStateBasedActions() {
             }
         }
 
+        if (!roleMap.isEmpty()) {
+            List<Set<Permanent>> rolesToHandle = roleMap.values()
+                    .stream()
+                    .map(Map::values)
+                    .flatMap(Collection::stream)
+                    .filter(s -> s.size() > 1)
+                    .collect(Collectors.toList());
+            if (!rolesToHandle.isEmpty()) {
+                for (Set<Permanent> roleSet : rolesToHandle) {
+                    int newest = roleSet.stream().mapToInt(Permanent::getCreateOrder).max().orElse(-1);
+                    // using the world enchantment rule for ties until otherwise corrected
+                    if (roleSet.stream().mapToInt(Permanent::getCreateOrder).filter(i -> i == newest).count() < 2) {
```

Comment:
I may be missing something, but what is that condition is checking exactly? Can there be more than 2 roles with the same createOrder?

Reply:
yeah I'm pretty sure it doesn't do anything and maybe it should just be removed

Reply:
Testing it, the role being sacrificed seems to be the newest one on all my attempts.
The one needing to be removed is the newest, the rest should be moved to the graveyard.

Reply:
this is fixed